soc: silabs: Update HAL for Series 2

asmellby opened the pull request:
Update HAL for Series 2 devices to Simplicity SDK 2024.6.

This HAL deprecates sl_device_init_emu(). Equivalent functions are now performed by sl_power_manager_init().

For now, soc.c remains compatible with both the Gecko SDK based HAL for Series 0 and 1, and the Simplicity SDK based HAL for Series 2. In a future commit, soc.c will be split between the two platforms.

asmellby committed:
soc: silabs: Update HAL for Series 2

Update HAL for Series 2 devices to Simplicity SDK 2024.6.

This HAL deprecates sl_device_init_emu(). Equivalent functions
are now performed by sl_power_manager_init().

For now, soc.c remains compatible with both the Gecko SDK based
HAL for Series 0 and 1, and the Simplicity SDK based HAL for
Series 2. In a future commit, soc.c will be split between the
two platforms.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

jhedberg commented:
@asmellby this seems to regress on my xg27 board. With current main I've able connect to it with the `peripheral_hr` app and browse the GATT services. With this PR I've able to discover the device, but connecting seems to hang forever. The exact build command I'm using:
```
west build -b xg27_dk2602a/efr32bg27c140f768im40 samples/bluetooth/peripheral_hr
```

jhedberg commented:
> this seems to regress on my xg27 board

One thing worth noting: what I physically have is a main board + radio board combo, however the build target is a dev kit - could that make a difference?

jhedberg commented:
Just tried with an xg24 radio board (for which there does exist an appropriate build target), and the result is the same. Output from the console looks like this:
```
*** Booting Zephyr OS build v3.7.0-3261-g0c62a2fa29c7 ***
[00:00:00.012,000] <wrn> bt_hci_core: Controller to host flow control not supported
[00:00:00.013,000] <wrn> bt_ecc: ECC HCI commands not available
[00:00:00.013,000] <inf> bt_hci_core: Identity: 80:4B:50:79:59:03 (public)
[00:00:00.013,000] <inf> bt_hci_core: HCI: version 5.4 (0x0d) revision 0x0000, manufacturer 0x02ff
[00:00:00.013,000] <inf> bt_hci_core: LMP: version 5.4 (0x0d) subver 0x0000
Bluetooth initialized
Starting Legacy Advertising (connectable and scannable)
Advertising successfully started
Checking LED device...done.
Configuring GPIO pin...done.
Start blinking LED...
Connected
Stop blinking LED.
Disconnected, reason 0x08
Starting Legacy Advertising (connectable and scannable)
Start blinking LED...
Connected
Stop blinking LED.
Disconnected, reason 0x08
Starting Legacy Advertising (connectable and scannable)
Start blinking LED...
Connected
```

asmellby committed:
drivers: bluetooth: silabs: Initialize PHY

Initialize PHY as part of controller init sequence.

Disable 2M and coded PHYs for now, as they cause issues on devices
that don't support them. This should be made configurable in the future.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

jhedberg reviewed (dismissed):
Seems to work fine now on my xg24. Thanks!
